content/docs/2.5/scalers/elasticsearch.md

```diff
@@ -0,0 +1,93 @@
++++
+title = "Elasticsearch"
+availability = "v2.5+"
+maintainer = "Community"
+description = "Scale applications based on elasticsearch search template query result."
+layout = "scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `elasticsearch` trigger that scales based on result of an [elasticsearch search template](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html) query. This lets you run searches without exposing Elasticsearch’s query syntax to the keda scaler.
+
+The trigger always requires the following information:
+
+```yaml
+triggers:
+  - type: elasticsearch
+    metadata:
+      addresses: "http://localhost:9200"
+      username: "elastic"
+      passwordFromEnv: "ELASTIC_PASSWORD"
+      index: "my-index"
+      searchTemplateName: "my-search-template-name"
+      params: "param1:value1;param2:value2"
+      valueLocation: "hits.total.value"
+      targetValue: "1"
+
+```
+
+**Parameter list:**
+
+- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
+- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional).
+- `username` - Username to authenticate with to Elasticsearch cluster.
+- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
+- `index` - Comma separated list of indexes to run the search template query on.
+- `searchTemplateName` - The search template name to run.
+- `params` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
+- `valueLocation` - [GJSON path notation](https://github.com/tidwall/gjson#path-syntax) to refer to the field in the payload containing the metric value.
+- `targetValue` - Target value to scale on. When the metric provided by the API is equal or higher to this value, KEDA will start scaling out. When the metric is 0 or less, KEDA will scale down to 0.
```

Comment:
Moved 2 entries to sort it more logically
```suggestion
- `addresses` - Comma separated list of hosts and ports of the Elasticsearch cluster client nodes.
- `username` - Username to authenticate with to Elasticsearch cluster.
- `passwordFromEnv` - Environment variable to read the authentication password from to authenticate with the Elasticsearch cluster.
- `searchTemplateName` - The search template name to run.
- `targetValue` - Target value to scale on. When the metric provided by the API is equal or higher to this value, KEDA will start scaling out. When the metric is 0 or less, KEDA will scale down to 0.
- `params` - Parameters that will be used by the search template. It supports multiples params separated by a semicolon character ( `;` )
- `valueLocation` - [GJSON path notation](https://github.com/tidwall/gjson#path-syntax) to refer to the field in the payload containing the metric value.
- `unsafeSsl` - Skip certificate validation when connecting over HTTPS. (Values: `true`, `false`, Default: `false`, Optional).
```

Reply:
looks like the `index` parameter has been removed?

Reply:
Woops, that's not intentional. I'd keep it before the search template name as you have suggested.